src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -224,16 +224,10 @@ private static Optional<LfAction> create(GeneratorAction action, LfNetwork lfNet
             OptionalDouble activePowerValue = action.getActivePowerValue();
             Optional<Boolean> relativeValue = action.isActivePowerRelativeValue();
             if (relativeValue.isPresent() && activePowerValue.isPresent()) {
-                double deltaTargetP;
-                if (relativeValue.get().equals(Boolean.TRUE)) {
-                    deltaTargetP = activePowerValue.getAsDouble() / PerUnit.SB;
-                } else {
-                    deltaTargetP = activePowerValue.getAsDouble() / PerUnit.SB - generator.getInitialTargetP();
-                }
-                var generatorChange = new GeneratorChange(generator, deltaTargetP);
+                var generatorChange = new GeneratorChange(generator, activePowerValue.getAsDouble() / PerUnit.SB, relativeValue.get());
                 return Optional.of(new LfAction(action.getId(), null, null, null, null, generatorChange, null, null));
             } else {
-                throw new UnsupportedOperationException("Generator action: configuration not supported yet.");
+                throw new UnsupportedOperationException("Generator action on " + action.getGeneratorId() + " : configuration not supported yet. Active power value and boolean for relative value are required.");
```

Comment:
The previous message was more correct: a generator action is really rich, we only support active power change, in an absolute way (only the double is required) or in a relative way (both double and boolean is required).

Reply:
I agree.
I suggest to delete the misleading addition and to keep the addition on the generator Id.
I have pushed a commit.